src/main/kotlin/org/rust/cargo/toolchain/tools/Cargo.kt

```diff
@@ -171,7 +172,14 @@ open class Cargo(toolchain: RsToolchain, useWrapper: Boolean = false)
     ): BuildScriptsInfo? {
         if (!isFeatureEnabled(RsExperiments.EVALUATE_BUILD_SCRIPTS)) return null
         val additionalArgs = listOf("--message-format", "json")
-        val commandLine = CargoCommandLine("check", projectDirectory, additionalArgs)
+        val nativeHelper = RsPathManager.procNativeHelper()
+        val envs = if (nativeHelper != null) {
+            EnvironmentVariablesData.create(mapOf(RUSTC_WRAPPER to nativeHelper.toString()), true)
+        } else {
+            EnvironmentVariablesData.DEFAULT
+        }
+
+        val commandLine = CargoCommandLine("check", projectDirectory, additionalArgs, environmentVariables = envs)
```

Comment:
Can we write a test to ensure that this command with our own `RUSTC_WRAPPER` doesn't actually compile target code? For example, check that we don't receive warning/error messages from the compiler, can extract build script info even there is a compilation error in some target code or anything else

Reply:
Added a test with a compilation error